guide/qip/qip-processor.rst

```diff
@@ -101,47 +181,222 @@ To let it find the optimal pulses, we need to give the parameters for :func:`qut
+**Output**
 
-.. image:: /gallery/qutip_examples/qip/images/sphx_glr_plot_qip_amplitude_noise_002.png
+.. testoutput::
 
-As the design of our simulator follows the physical realization, so is the noise simulation. Noise can be added to the processor at different levels:
+    [0, 1, 3, 2, 2, 3, 4]
+
+The result shows the scheduling order of each gate in the original circuit.
+
+For pulse schedule, or scheduling gates with different duration,
+one will need to wrap the :class:`qutip.qip.circuit.Gate` object with :class:`qutip.qip.compiler.instruction` object,
+with a parameter `duration`.
+The result will then be the start time of each instruction.
+
+Noise Simulation
+----------------
+
+In the common way of QIP simulation, where evolution is carried out by gate matrix product, the noise is usually simulated with bit flipping and sign flipping errors.
+The typical approaches are either applying bit/sign flipping gate probabilistically
+or applying Kraus operators representing different noisy channels (e.g. amplitude damping, dephasing) after each unitary gate evolution. In the case of a single qubit, they have the same effect and the parameters in the Kraus operators are exactly the probability of a flipping error happens during the gate operation time.
+
+Since the processor simulates the state evolution at the level of the driving Hamiltonian, there is no way to apply an error operator to the continuous-time evolution. Instead, the error is added to the pulses (coherent control error) or the collapse operators (Lindblad error) contributing to the evolution. Mathematically, this is no different from adding error channel probabilistically (it is actually how :func:`qutip.mcsolve` works internally). The collapse operator for single-qubit amplitude damping and dephasing are exactly the destroying operator and the sign-flipping operator. One just needs to choose the correct coefficients for them to simulate the noise, e.g. the relaxation time T1 and dephasing time T2. Because it is based on the open system evolution instead of abstract operators, this simulation is closer to the physical implementation and requires less pre-analysis of the system.
+
+Compared to the approach of Kraus operators, this way of simulating noise is more computationally expensive. If you only want to simulate the decoherence of single-qubit relaxation and the relaxation time is much longer than the gate duration, there is no need to go through all the calculations. However, this simulator is closer to the real experiment and, therefore, more convenient in some cases, such as when coherent noise or correlated noise exist. For instance, a pulse on one qubit might affect the neighbouring qubits, the evolution is still unitary but the gate fidelity will decrease. It is not always easy or even possible to define a noisy gate matrix. In our simulator, it can be done by defining a :class:`~qutip.qip.noise.ControlAmpNoise` (Control Amplitude Noise).
```

Comment:
seems odd that the class isn't just called `ControlAmplitudeNoise` to me, but ok!